spec/models/atmosphere/user_fund_spec.rb

```diff
@@ -33,4 +33,75 @@
         to include(I18n.t('funds.unique_user'))
   end
 
+
+  describe '#ensure_single_default' do
+
+    it 'makes first user_fund default' do
+      expect(user_fund.default).to be_truthy
+    end
+
+    it 'blocks setting the only fund as not default' do
+      expect(user_fund.default).to be_truthy
+      user_fund.default = false
+      user_fund.save
+      expect(user_fund.reload.default).to be_truthy
+    end
+
+    it 'does not bother single default fund creation' do
+      pc = create(:poor_chap)
+      uf = create(:user_fund, user: pc, default: true)
+      expect(uf.reload.default).to be_truthy
+    end
+
+    it 'keeps only single default user fund' do
+      uf = create(:user_fund, user: user_fund.user, default: true)
+      expect(Atmosphere::UserFund.count).to eq 2
+      expect(uf.reload.default).to be_truthy
+      expect(user_fund.reload.default).to be_falsey
+    end
+
+    it 'does not prematurely default to an invalid fund' do
+      begin
+        create(:user_fund,
+               fund: user_fund.fund,
+               user: user_fund.user,
+               default: true)
+      rescue ActiveRecord::RecordInvalid
+      end
+      expect(Atmosphere::UserFund.count).to eq 1
+      expect(user_fund.reload.default).to be_truthy
+    end
+  end
+
+
+  describe '#reallocate_default' do
+
+    it 'does nothing when last fund is removed' do
+      user_fund.destroy
+      expect(Atmosphere::UserFund.count).to eq 0
+    end
+
+    it 'reallocates removed default' do
+      uf = create(:user_fund, user: user_fund.user)
+      uf2 = create(:user_fund, user: user_fund.user)
```

Comment:
The is already spec checking if only one user fund can be set to default. As a conclusion maybe only one additional user fund (`uf`) can be created here (and the same situation in spec below) to check reallocation conditions?